File: VB/VB2/WebService1.asmx.vb
Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports System.ComponentModel
Imports System.Data.SqlClient

' To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line.
<System.Web.Script.Services.ScriptService()>
<System.Web.Services.WebService(Namespace:="http://tempuri.org/")> _
<System.Web.Services.WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)>
<ToolboxItem(False)>
Public Class WebService1
    Inherits System.Web.Services.WebService

    <WebMethod()>
    Public Function HelloWorld() As String
        Return "Hello World"
    End Function


    <WebMethod()>
    Public Function SearchFoods(ByVal prefixText As String, ByVal count As Integer, ByVal contextKey As Integer) As List(Of String)
        Dim lstfoods As List(Of String) = New List(Of String)
        Dim connectionstring As String = "Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=DataStore;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False"
        Using conData As SqlConnection = New SqlConnection(connectionstring)
            conData.Open()
            'Dim strSQL As String = "EXEC kd_selFoodItemsSearch @OrgID,@SearchText ='" & prefixText & "'"
            Dim strSQL As String = "EXEC kd_selFoodItemsSearch @OrgID,@SearchText"
            Dim cmd As SqlCommand = New SqlCommand
            cmd.CommandText = strSQL
            cmd.Parameters.AddWithValue("@OrgID", contextKey)
            cmd.Parameters.AddWithValue("@SearchText", prefixText)
            cmd.Connection = conData
            Dim sdr As SqlDataReader = cmd.ExecuteReader
            While sdr.Read
                lstfoods.Add(sdr("FoodItem").ToString)
            End While
            sdr.Close()
        End Using

        Return lstfoods
    End Function
End Class

File: WebApplication0419Home/WebApplication0419Home/WebService1.asmx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Services;

namespace WebApplication0419Home
{
    /// <summary>
    /// WebService1 的摘要说明
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    // 若要允许使用 ASP.NET AJAX 从脚本中调用此 Web 服务，请取消注释以下行。 
    [System.Web.Script.Services.ScriptService]
    public class WebService1 : System.Web.Services.WebService
    {

        [WebMethod]
        public string HelloWorld()
        {
            return "Hello World";
        }
        [WebMethod]
        public List<string> GetData(string prefixText)
        {
            List<string> data = new List<string>();
            data.Add("abr,wuxi,1@qq,2");
            data.Add("abc,shanghai,2@cc,3");
            data.Add("abraham,beijing,5@qi,6");
            data.Add("ab,wuxi,3@qq,9");
            data.Add("d,shenzhen,2@qq,5");
            data.Add("ad,shanghai,4@qy,5");
            data.Add("adc,beijing,5@yy,6");
            data.Add("ap,shanghai,7@yy,8");
            data.Add("apc,wuxi,5@yy,9");
            data.Add("fa,wuxi,5@qq,9");
            data.Add("fat,shagnhai,2@qq,8");
            return data.Where(p => p.StartsWith(prefixText)).ToList();
        }

    }
}
